earth_enterprise/src/common/khConfigFileParser.cpp

```diff
@@ -0,0 +1,137 @@
+#include "khConfigFileParser.h"
+#include <algorithm>
+#include <regex>
+#include <fstream>
+#include "common/khThread.h"
+
+FileNotPresentException::FileNotPresentException(const std::string& fn)
+{
+    str = fn + " is not present";
+}
+
+OptionsEmptyException::OptionsEmptyException()
+{
+	str = "no list of options present";
+}
+
+ValueNotPresentException::ValueNotPresentException(const std::string& key)
+{
+    str = key + " has an unset value";
+}
+
+KeyNotPresentException::KeyNotPresentException(const std::string& key)
+{
+    str = key + " is not a valid key";
+}
+
+void khConfigFileParser::addOption(std::string option) 
+{
+    // if empty, return
+    if (option.size() == 0) return;
+    std::transform(option.begin(), option.end(),
+                   option.begin(), ::toupper);
+    options.push_back(option);
+}
+    
+
+bool khConfigFileParser::sanitize(std::string& line)
+{
+    // remove whitespace
+    line = std::regex_replace(line, std::regex("\\s+"),"");
+    
+    // remove comments
+    auto idx = line.find('#');
+    if (idx != std::string::npos) 
+    {
+        line.erase(line.begin()+idx,line.end());
+    }
+    
+    // remove empty lines
+    if (!line.size()) return false;
+    
+    
+    // make uppercase
+    std::transform(line.begin(), line.end(),
+                   line.begin(), ::toupper);
+    return true;
+}
+
+
+void khConfigFileParser::split(const std::string& line, std::string& key, std::string& value)
+{
+    std::regex eq("=");
+    std::vector<std::string> splitLine
+    {
+        std::sregex_token_iterator(line.begin(),line.end(),eq,-1),
+        {}
+    };
+    key = splitLine[0];
+    if (splitLine.size() == 2)
+    {
+        value = splitLine[1];
+    }
+}
+
+bool khConfigFileParser::isKeyPresent(const std::string& key)
+{
+    for (const auto& a : options)
+    {
+        if (a == key) return true;
+    }
+    return false;
+}
+
+static khMutexBase xmlParmsLock = KH_MUTEX_BASE_INITIALIZER;
+
+void khConfigFileParser::parse(const std::string& fn)
+{
+	std::ifstream file;
```

Comment:
You've got some funny indentation here (not consistent in this function. It would be nice to clean that up.